bika/lims/browser/widgets/referencewidget.py

```diff
@@ -198,73 +196,106 @@ def get_editbutton_options(self):
 registerWidget(ReferenceWidget, title='Reference Widget')
 
 class ajaxReferenceWidgetSearch(BrowserView):
-
     """ Source for jquery combo dropdown box
     """
 
-    def __call__(self):
-        plone.protect.CheckAuthenticator(self.request)
-        page = self.request['page']
-        nr_rows = self.request['rows']
-        sord = self.request['sord']
-        sidx = self.request['sidx']
-        colModel = json.loads(_u(self.request.get('colModel', '[]')))
-        discard_empty = json.loads(_c(self.request.get('discard_empty', "[]")))
-        rows = []
+    @property
+    def num_page(self):
+        """Returns the number of page to render
+        """
+        num_page = self.request.get("page", 1)
+        if api.is_floatable(num_page) and int(num_page) > 0:
+            return int(num_page)
+        return 1
+
+    @property
+    def num_rows_page(self):
+        """Returns the number of rows per page to render
+        """
+        num_rows_page = self.request.get("rows", 10)
+        if api.is_floatable(num_rows_page) and int(num_rows_page) > 0:
+            return int(num_rows_page)
+        return 10
+
+    def get_field_names(self):
+        """Return the field names to get values for
+        """
+        col_model = _u(self.request.get("colModel", "[]"))
```

Comment:
Optional;)

We should avoid this pattern with the JSON default value and return directly if `self.request.get("colModel") is None`.
 The reason is that if the `colModel` key is found in the request but does not contain a list, this code fails. Therefore, another check would make it more solid, e.g. `isinstance(col_model, (list, tuple))` or add an API function `to_list`.